tests/foreman/ui/test_organization.py

```diff
@@ -365,13 +382,15 @@ def test_positive_remove_domain(self):
 
         :CaseLevel: Integration
         """
+
         with Session(self.browser) as session:
+            org_name = _org_name_module_scope(session)
```

Comment:
I like your implementation as it is clearer than current `set_session_org`, my suggestion s to bind the cache getter to the session.

instead of:

```python
org_name = _org_name_module_scope(session)
```

we do:

```python
org = session.get_org() 
org_name = session.get_org_name() 
```

and this uses the same `_org_name_module_scope` cache factory.